### DISCUSSION TITLE
Step size limit

### DISCUSSION THREAD

**Discussion by eytanadler** [coin-or/Ipopt#778] [Ideas]
I'm using Ipopt in the quasi-Newton mode and finding that it often takes such large steps in the first few iterations that my objective function cannot be solved at the requested design variables. Presumably, this is because the Hessian does not yet accurately represent the design space.

In SNOPT, there is a phenomenally useful option called the `Major step limit`, which, in rough terms, puts an upper bound on the change in the design variable vector for a given iteration. I posted the [SNOPT manual](https://ccom.ucsd.edu/reports/UCSD-CCoM-19-01.pdf)'s description of this parameter at the bottom of this discussion. I have used this parameter in the past to dramatically improve robustness. It also helps the optimizer converge to the nearest optimum to the initial design.

I have not been able to find a similar option in Ipopt, though there does seem to have been [some discussion](https://github.com/coin-or/Ipopt/discussions/675) about it. Is this a feature that could be easily added?

The `Major step limit` description in the SNOPT manual:
<img width="1298" alt="image" src="https://github.com/coin-or/Ipopt/assets/63426601/06e51b12-6c21-4ff7-9625-7cbf4ce03679">

**Comment by svigerske** [comment]
I think that this is still not available, but we can keep it in mind to add this. I think it needs modifications around

https://github.com/coin-or/Ipopt/blob/7c95c3b4fdb018f73db40ef20282aacdc9e877a9/src/Algorithm/IpBacktrackingLineSearch.cpp#L685-L699

**Reply by eytanadler** [reply]
Ok, thanks! If I have a chance, I might take a look and try to figure it out. Either way, it'd be a great feature to have
